File: tests/journal.lib/chronicler_device.cc
// -*- c++ -*-
//
// michael a.g. aïvázis <michael.aivazis@para-sim.com>
// (c) 1998-2023 all rights reserved


// get the journal
#include <pyre/journal.h>
// support
#include <cassert>


// aliases
using trash_t = pyre::journal::trash_t;
using chronicler_t = pyre::journal::chronicler_t;


// exercise getting/setting the global default device
int main() {
    // get the default device
    auto builtin = chronicler_t::device();

    // make a new device
    auto custom = std::make_shared<trash_t>();
    // install it
    chronicler_t::device(custom);

    // check that the current device is the one we just installed
    assert (chronicler_t::device().get() == custom.get());

    // all done
    return 0;
}


// end of file

File: tests/journal.pkg/chronicler_device.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis <michael.aivazis@para-sim.com>
# (c) 1998-2023 all rights reserved


# access to the global device
def test():
    """
    Exercise getting/setting the global device
    """
    # get the chronicler
    from journal.Chronicler import Chronicler as chronicler
    # and the trash can
    from journal.Trash import Trash as trash

    # make a new device
    custom = trash()
    # install it
    chronicler.device = custom

    # verify that it was installed correctly
    assert chronicler.device is custom

    # all done
    return


# main
if __name__ == "__main__":
    # prohibit the journal bindings
    journal_no_libjournal = True
    # run the test
    test()


# end of file
